pkg/crypto/certmanager/manager.go

```diff
@@ -264,12 +264,19 @@ func (cm *CertificateManager) checkRenew() time.Duration {
 		// if it needs to be renewed.
 		if cert, err := readCert(cm.storer); err != nil {
 			cm.logger.Println(err)
-		} else { // Make use of a certificate, even if expired, then rewnew.
+		} else { // Make use of newer certificate, even if expired, then rewnew.
 			cm.rwMutex.Lock()
 			if cm.certificate == nil {
 				go cm.fileWrite(cert)
+				cm.certificate = cert
+			} else if cert.notAfter.After(cm.certificate.notAfter) {
```

Comment:
why not make this part of an or condition on the if on line 269?

Reply:
Done.